[qtwebengine] allow building with MSVC 14.5 / Visual Studio 2026

rokm opened the pull request:
Trying to build QtWebEngine with MSVC Toolset 14.5 / Visual Studio 2026 currently fails with "Build requires Visual Studio 2019 or higher." due to explicit check for toolset versions 14.2 and 14.3. Extending the check for 14.5 fixes the problem.

Upstream bug: https://qt-project.atlassian.net/browse/QTBUG-142323

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

rokm committed:
[qtwebengine] allow building with MSVC 14.5 / Visual Studio 2026

rokm commented:
@microsoft-github-policy-service agree

nickdademo commented:
@rokm Out of curiosity, could you build `qtwebengine` with VS2022? - I never could: https://github.com/microsoft/vcpkg/issues/46606

rokm commented:
> @rokm Out of curiosity, could you build `qtwebengine` with VS2022? - I never could: #46606

Yes, I've previously built `qtwebengine` 6.8.2 and 6.8.3 with VS 2022.

Now that you mention it, on the very first attempt at building 6.8.2, I ran into a similar issue with `ACTION //components/resources/ssl/ssl_error_assistant:make_ssl_error_assistant_protobuf` failing. 

In my case, the symptoms and the underlying cause were as follows:

- the python process that was supposed to generate the protobuf was crashing   when it was trying to load `_ctypes` extension. This could also be reproduced by simply running the python from virtual environment, and trying to do `import ctypes`.

- the reason for crash was that vcpkg installed python 3.12.7 for virtual environment, but I also had python 3.12.1 installed in `%LOCALAPPDATA%\Programs\Python\Python312`, and for some reason, extension from this installation's DLL directory ended up being loaded. And apparently due to being of older version, it caused the crash.

- it's worth noting that the afore-mentioned 3.12.1 python installation was *not* in `PATH`, but still its directories seemed to be added to the `sys.path` of the python from virtual environment. But even so, the order of paths in `sys.path` should have ensured that the correct extension from 3.12.7 DLL directory was loaded; and this was the case for other extensions...

- so looking at the `_ctypes` extension file in the 3.12.7 DLL directory, it turned out to have size 0. I have no idea how that came to be, but my primary suspect was (and still is) Windows Defender.

So for my next attempt, I added the build hard drive (where vcpkg directory, its build tree directory, and package cache directory, reside) to Windows Defender's exempted paths. For good measure, I also temporarily renamed `%LOCALAPPDATA%\Programs\Python`. Then cleaned up all vcpkg-related directories, tried again, and the 6.8.2 was built successfully.

When building 6.8.3, I didn't rename `%LOCALAPPDATA%\Programs\Python` anymore (I had updated the python 3.12 installation there to the latest version, though). But the Windows Defender exemption was still in place. It also succeeded.

Last weekend, I've decided to set up the build environment on my new Windows 11 installation, which now has VS 2026. Windows Defender exemption is still in place. But this system has only python 3.13 installed in `%LOCALAPPDATA%\Programs\Python` at the moment. After applying patch in this PR, the build of `qtwebengine` 6.9.2 also succeeded.

BillyONeal reviewed (approved):
Thanks!
